packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx

```diff
@@ -224,12 +236,14 @@ function GridColumnsPanel(props: GridColumnsPanelProps) {
         <rootProps.components.BaseButton
           onClick={() => toggleAllColumns(false)}
           {...rootProps.componentsProps?.baseButton}
+          disabled={disableHideAllButton}
```

Comment:
These new props (disableHideAllButton, disableShowAllButton) should override `componentsProps.baseButton.disabled` as it's generic for base button, but the new props are explicit to column panel buttons.

Reply:
It seems better to hide both if both props are passed, I'll update it.